core/baseline-core-v1.0-psd01.md

```diff
@@ -191,8 +191,273 @@ The ability of a Party operating Workflows on a baseline-compliant implementatio
 
 # 2 Design and Architecture
 
+This section provides definitions, key concepts, and overviews of the components of a Baseline Protocol Implementation compliant with the requirements of this document. 
+
+## 2.1 Agreement
+An Agreement between two or more Counterparties allows for transactions between them dealing with, for example, commercial items such as products. An Agreement governs and defines all transactions between counterparties.
+
+**[R1]**	Transacting counterparties  MUST have a corresponding agreement.
+
+## 2.2	State Object
+A State Object is an item which can be exchanged between counterparties to an agreement, whose state the counterparties have agreed on and which is defined in an agreement. In the context of this standard, a State Object is assumed to represent either an asset, a product offering or a service offering transacted between counterparties.
+Examples include but are not limited to:
+* Digital Services
+* Physical Products
+* Financial Assets
+ 
+## 2.3 Transacting Counterparties
+A transaction counterparty, or simply counterparty, that requests one or more state objects from another counterparty is the Requester with respect to the state object.  The Requester can also be the end (non-counterparty) recipient of one or more state objects.
```

Comment:
Do we have an actual requester role? In the current structure, a Provider (also has been referred to as Sender) sends a record to a counterparty. afaik, there is no current or planned way for a Requester to initiate the process, though I do see the role appearing in cases where a counterparty's record has become "debased" (no longer consistent with proof) and the counterparty must "request" the original record from one or more of the other counterparties.

Reply:
@humbitious ... we do ... you request a service for example when you create an order. More generally, the Requester requests from the Provider the agreement to a suggested (commercial) state change on one or more (commercial) state objects based on a commercial document.

Reply:
We should discuss this. I've never seen anything in our work that would constitute a Requester role. I understand the concept being applied here, but I'd like to get more clarity on this one.

Reply:
I think Provider and Requester are more clear in terms of the non-repudiation feature.